tests/e2e/kafkatopic_webhook.go

```diff
@@ -0,0 +1,318 @@
+			// Example error:
+			// The KafkaTopic "topic-test-internal" is invalid:
+			// * spec.partitions: Invalid value: 0: number of partitions must be larger than 0 (or set it to be -1 to use the broker's default)
+			// * spec.replicationFactor: Invalid value: 0: replication factor must be larger than 0 (or set it to be -1 to use the broker's default)
+			Expect(len(strings.Split(output, "\n"))).To(Equal(3))
+			Expect(output).To(And(
+				ContainSubstring("spec.partitions: Invalid value: %s: number of partitions must be larger than 0 (or set it to be -1 to use the broker's default)", "0"),
+				ContainSubstring("spec.replicationFactor: Invalid value: %s: replication factor must be larger than 0 (or set it to be -1 to use the broker's default)", "0")))
+		})
+
+		// In the current validation webhook implementation, this case can only be encountered on a Create operation
+		It("Test ReplicationFactor larger than number of brokers", func() {
+			output, err := applyK8sResourceFromTemplateWithDryRun(kubectlOptions,
+				kafkaTopicTemplate,
+				map[string]interface{}{
+					"Name":              testInternalTopicName,
+					"TopicName":         testInternalTopicName,
+					"ReplicationFactor": "10",
+				},
+				dryRunStrategyServer,
+			)
+			Expect(err).To(HaveOccurred())
+			// Example error:
+			// The KafkaTopic "topic-test-internal" is invalid: spec.replicationFactor: Invalid value: 10: replication factor is larger than the number of nodes in the kafka cluster (available brokers: 3)
+			Expect(len(strings.Split(output, "\n"))).To(Equal(1))
+			Expect(output).To(And(
+				ContainSubstring("spec.replicationFactor: Invalid value"), ContainSubstring("replication factor is larger than the number of nodes in the kafka cluster")))
+		})
+
+		// Test case involving existing CRs but not necessarily an Update operation
+		When("Testing conflicts similar CRs", func() {
+			requireDeployingKafkaTopic(kubectlOptions, testInternalTopicName)
+
+			It("Testing conflict on spec.name", func() {
+				By("With managedBy koperator annotation")
+				output, err := applyK8sResourceFromTemplateWithDryRun(kubectlOptions,
+					kafkaTopicTemplate,
+					map[string]interface{}{
+						"Name":        testInternalTopicName + "-different-cr-name",
+						"TopicName":   testInternalTopicName,
+						"Namespace":   kubectlOptions.Namespace,
+						"Annotations": []string{"managedBy: koperator"},
+					},
+					dryRunStrategyServer,
+				)
+				Expect(err).To(HaveOccurred())
+				// Example error:
+				// The KafkaTopic "topic-test-internal-different-cr-name" is invalid: spec.name: Invalid value: "topic-test-internal": kafkaTopic CR 'topic-test-internal' in namesapce 'kafka' is already referencing to Kafka topic 'topic-test-internal'
+				Expect(len(strings.Split(output, "\n"))).To(Equal(1))
+				Expect(output).To(
+					ContainSubstring("spec.name: Invalid value: %[1]q: kafkaTopic CR '%[1]s' in namesapce '%[2]s' is already referencing to Kafka topic '%[1]s'",
```

Comment:
Suggestion: We have a typo in the log message in our source code: https://github.com/banzaicloud/koperator/blob/master/pkg/webhooks/kafkatopic_validator.go#L256

It'd be great if we can fix that typo in the implementation, and update the tests accordingly.

```suggestion
					ContainSubstring("spec.name: Invalid value: %[1]q: kafkaTopic CR '%[1]s' in namespace '%[2]s' is already referencing to Kafka topic '%[1]s'",
```

Reply:
The typo in the tests is, indeed, on purpose.  
I already tried fixing the "root" typo in the implementation (in a commit that I overwrote in the mean time) so that I wouldn't have to have a typo on purpose in the tests but the CI was failing on that typo as if the fixed webhook code wasn't being used.

I can give it another go and report back.

Reply:
We talked about it offline and updating the webhook itself and the e2e test about in one go is not possible. The middle ground is to push to fix for the typo in the webhook itself in this PR. The update to the e2e test would probably need to come after releasing a new version.

The fixed typo in the webhook is in 2a57f9acae56f55e9156ffdc3f8bccf7b56aca80.